acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,94 @@
+
+  step 'On master, create new environments that will result in Failed' do
+    site_manifest = File.join(environmentpath, tmp_environment, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  fail('intended failure')
+}
+SITEPP
+    site_manifest2 = File.join(environmentpath, tmp_environment2, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest2, <<-SITEPP)
+node default {
+  exec{'/bin/false':}
+}
+SITEPP
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent),
+                         pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  target_identities = []
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment}" do
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment} environment")
+        assert_equal('failed', action_result['status'],
+                     "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: "\
+                     "'#{action_result["status"]}'")
+      end
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment2}" do
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment2,
```

Comment:
This doesn't work for me:
    [root@rvmgw0nbcg7tghl ~]# puppet agent -t --environment environment2
    Info: Using configured environment 'environment2'
    Info: Retrieving pluginfacts
    Info: Retrieving plugin
    Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Could not parse for environment environment2: Permission denied - /etc/puppetlabs/code/environments/environment2/manifests/site.pp on node rvmgw0nbcg7tghl.delivery.puppetlabs.net
    Warning: Not using cache on failed catalog
    Error: Could not retrieve catalog; skipping run

I think permissions need modified for site.pp after using create_remote_file to create it
chmod 644 site.pp results in:

    [root@rvmgw0nbcg7tghl ~]# puppet agent -t --environment environment2
    Info: Using configured environment 'environment2'
    Info: Retrieving pluginfacts
    Info: Retrieving plugin
    Info: Caching catalog for rvmgw0nbcg7tghl.delivery.puppetlabs.net
    Info: Applying configuration version '1462359992'
    Error: /bin/false returned 1 instead of one of [0]
    Error: /Stage[main]/Main/Node[default]/Exec[/bin/false]/returns: change from notrun to 0 failed: /bin/false returned 1 instead of one of [0]
    Notice: Applied catalog in 0.02 seconds